components/calendar/src/cal/chinese.rs

```diff
@@ -110,61 +110,66 @@ pub trait Rules: Clone + core::fmt::Debug + crate::cal::scaffold::UnstableSealed
-/// or from sources like the excellent compilation by
-/// [Yuk Tung Liu](https://ytliu0.github.io/ChineseCalendar/table.html).
+/// For years since 1912, this uses the [GB/T 33661-2017] rules.
+/// As accurate computation is computationally expensive, years until
+/// 2100 are precomputed. If performance beyond 2100 is an issue, clients
+/// can implement their own [`Rules`] type containing more precomputed data.
+/// We note that the calendar is inherently uncertain for some future dates.
 ///
-/// Dates before 1901 are mainly of historical interest, and were calculated
-/// using different [methods](https://ytliu.epizy.com/Shixian/Shixian_summary.html)
-/// than [GB/T 33661-2017]. These algorithms and data are not included for binary
-/// size reasons. If accuracy is required in this range clients can implement their
-/// own [`Rules`] type using the desired data.
+/// Before 1912 [different rules](https://ytliu.epizy.com/Shixian/Shixian_summary.html)
+/// were used. This type produces correct data for the years 1900-1912, and
+/// falls back to [GB/T 33661-2017] proleptically before 1900. If accuracy
+/// is required before 1900, clients can implement their own [`Rules`] type
+/// using data such as from the excellent compilation by [Yuk Tung Liu].
 ///
-/// [GB/T 33661-2017]: https://openstd.samr.gov.cn/bzgk/gb/newGbInfo?hcno=E107EA4DE9725EDF819F33C60A44B296
+/// [Purple Mountain Observatory for the years 1900-2025]: http://www.pmo.cas.cn/xwdt2019/kpdt2019/202203/P020250414456381274062.pdf
+/// [Hong Kong Observatory for the years 1901-2100]: https://www.hko.gov.hk/en/gts/time/conversion.htm
+/// [GB/T 33661-2017]: China::gb_t_33661_2017
+/// [Yuk Tung Liu]: https://ytliu0.github.io/ChineseCalendar/table.html
 #[derive(Copy, Clone, Debug, Default)]
 #[non_exhaustive]
 pub struct China;
 
+impl China {
+    /// Computes [`LunarChineseYearData`] according to [GB/T 33661-2017],
+    /// as implemented by [`calendrical_calculations::chinese_based::Chinese`].
+    ///
+    /// The rules specified in [GB/T 33661-2017] have only been used
+    /// since 1912, applying them proleptically to years before 1912 will not
+    /// necessarily match historical calendars.
+    ///
+    /// Note that for future years there is a small degree of uncertainty, as
+    /// [GB/T 33661-2017] depends on the uncertain future [difference between UT1
+    /// and UTC](https://en.wikipedia.org/wiki/Leap_second#Future).
+    /// As noted by
+    /// [Yuk Tung Liu](https://ytliu0.github.io/ChineseCalendar/computation.html#modern),
+    /// years as early as 2057, 2089, and 2097 have lunar events very close to
+    /// local midnight, which might affect the start of a (single) month if additional
+    /// leap seconds are introduced.
+    ///
+    /// [GB/T 33661-2017]: https://openstd.samr.gov.cn/bzgk/gb/newGbInfo?hcno=E107EA4DE9725EDF819F33C60A44B296
+    pub fn gb_t_33661_2017(related_iso: i32) -> LunarChineseYearData {
```

Comment:
observation: new public API

Reply:
It's not, it was on `LunarChineseYearData` before, but it's China-specific so I moved it here